helix-rest/src/main/java/org/apache/helix/rest/server/resources/helix/PropertyStoreAccessor.java

```diff
@@ -0,0 +1,68 @@
+package org.apache.helix.rest.server.resources.helix;
+
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+import java.io.UnsupportedEncodingException;
+import java.net.URLDecoder;
+import java.nio.charset.StandardCharsets;
+import java.util.Collections;
+
+import javax.ws.rs.GET;
+import javax.ws.rs.Path;
+import javax.ws.rs.PathParam;
+import javax.ws.rs.QueryParam;
+import javax.ws.rs.core.Response;
+
+import org.apache.helix.AccessOption;
+import org.apache.helix.HelixException;
+import org.apache.helix.PropertyPathBuilder;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.manager.zk.ZkBaseDataAccessor;
+import org.apache.helix.store.HelixPropertyStore;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+@Path("/clusters/{clusterId}/propertyStore")
+public class PropertyStoreAccessor extends AbstractHelixResource {
+  private static Logger LOG = LoggerFactory.getLogger(PropertyStoreAccessor.class);
+
+  @GET
+  public Response getPropertyByPath(@PathParam("clusterId") String clusterId,
+      @QueryParam("path") String urlEncodedPath) {
```

Comment:
I'm thinking that for the ease of use from the user's perspective, something like this might be cleaner:

Making the path a PathParam instead of a QueryParam (since we are really querying for an entity, path param makes sense):


 ```
  @GET
  @PathParam("{path : .+}") // escapes slashes
  public Response getPropertyByPath(@PathParam("clusterId") String clusterId,
      @PathParam("path") String znodePath)
```

What do you think? In this case, the user can directly pass in the path as in

`http://endpoint.linkedin.biz/clusters/CLUSTER/propertyStore/PATH
`
But we'd need to review what we define PATH should be. Should it be the complete path? or the path under /PROPERTYSTORE/? That would be a good thing to discuss.

Reply:
It's the first idea for me as well. However, realizing the PATH parameter can be something nasty like "a/b/c/d", the url becomes unnatural `"http://endpoint.linkedin.biz/clusters/CLUSTER/propertyStore/a/b/c/d"`. Something like this will just confuse people in practice. The URL encode is natural to HTTP request to escape the '/'